packages/blade/src/components/Counter/_decisions/decisions.md

```diff
@@ -0,0 +1,46 @@
+# Counter 
+
+Counters are visual indicators that contains numerical values, tallies or counts in regards to some context.
+
+<img src="./counter-thumbnail.png" width="380" />
+
+## Design
+
+[Figma Link](https://www.figma.com/file/jubmQL9Z8V7881ayUD95ps/Blade---Payment-Light?node-id=8222%3A70827) to all variants of the Counter component
+
+## API
+
+| Prop | Type | Default | Description | Required |
+|---|---|---|---|---|
+| value | `number` | `undefined` | The numeric value of the counter which will be shown. | ✅ |
+| max | `number` | `undefined` | The max value of the counter, if the `value` exceededs `max` then the counter will show a `+` at the end. |  |
+| intent | `positive`,`negative`, `notice`, `information`, `neutral` | `neutral` | The variant of the counter to be rendered. |  |
+| contrast | `low`, `high` | `low` | The contrast of the counter to be rendered. |  |
+
+## Usage Example
+
+```tsx
+import { Counter } from '@razorpay/components';
+
+<Counter 
+  value={150}
```

Comment:
why `value` as a prop and not a children similar to Typography components?

Reply:
_2 reasons:_ 

1. The counters will accept numbers only so the value have to be set to `number`
 
Setting the `children: number` the consumers will have to use JSX expressions to pass the number in children which looks kinda odd, otherwise TS will yell. 

```tsx
<Counter>{20}</Counter>

<Counter>20</Counter> // <-- error
```

2. When the `max` prop is set we also need to append `+` at the end of the `value`. If we use `children` intuitively the expectation might be that whatever we put in the `children` will be rendered but in reality we will modify what's being rendered. 

```tsx
// renders `10+`
<Counter max={10}>{20}</Counter>

// vs

// renders `10+` similar API to `<input value={} max={} />`
<Counter value={20} max={10} />  
```